src/StochasticStyles/spawning.jl

```diff
@@ -266,22 +261,27 @@ struct WithoutReplacement{T} <: SpawningStrategy
     WithoutReplacement(threshold::T=0.0) where {T} = new{T}(threshold)
 end
 
-@inline function spawn!(s::WithoutReplacement, w, offdiags::AbstractVector, add, val, boost=1)
+@inline function spawn!(s::WithoutReplacement, w, column, val, boost=1)
     spawns = zero(valtype(w))
     num_attempts = max(floor(Int, abs(val) * boost), 1)
 
     if abs(num_attempts) ≤ 1
-        spawn!(SingleSpawn(s.threshold), w, offdiags, add, val)
+        spawn!(SingleSpawn(s.threshold), w, column, val)
     else
         magnitude = val / num_attempts
 
-        num_offdiags = length(offdiags)
+        num_offdiags = num_offdiagonals(column)
         prob = 1 / num_offdiags
-
+        offdiags = offdiagonals(column)
+        if !(offdiags isa AbstractVector)
+            throw(ArgumentError(
+                "The WithoutReplacement strategy requires offdiagonals to be an AbstractVector. Please use a different strategy."
+            ))
+        end
```

Comment:
Hmm, I think I would put this at the beginning of the function body, in the hope that the compiler can resolve the `if` statement at compile time. The way it is written now, some runtime information is required to decide whether to throw this error and it becomes a branch in the code that should be avoided. You also wouldn't want you code to error halfway through the calculation when`num_attempts` happens to grow beyond 1.

Another option (to ensure the compiler action happens) would be to put a function barrier in and pass the `offdiagonals` to the inner function.